pinot-ingestion-jobs/pinot-standalone/src/main/java/org/apache/pinot/ingestion/common/PinotFSSpec.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.ingestion.common;
+
+import java.util.Map;
+
+
+public class PinotFSSpec {
+
+  String _scheme;
+
+  String _className;
+
+  Map<String, String> _configs;
+
+  public String getScheme() {
+    return _scheme;
+  }
+
+  public void setScheme(String scheme) {
+    _scheme = scheme;
+  }
+
+  public String getClassName() {
+    return _className;
+  }
+
+  public void setClassName(String className) {
+    _className = className;
+  }
+
+  public Map<String, String> getConfigs() {
```

Comment:
Presumably these are File system specific. Should they be Properties instead of Map (as in the init() call to a PinotFS)?

Reply:
Can we represent properties in a yaml file? will the yaml/json serializers/deserializers work without issues if we use Properties instead of Map. If yes, we can use properties

Reply:
My personal bias is towards NOT using auto-conversion to keep library compatibility, etc. I understand the intention of not having boiler plate code, and too many hard-coded values, but we have been bitten BADLY by incompatible enhancements to these packages, and if there are multiple of them (now yaml as well as json) then we better stick with packages that have a good reputation to keep method signatures etc. backward compatible.

That said, I am not sure how to work with yaml trying to represent arbitrary properties that a file system may have. e.g.

hadoopfs.baseurl
hadoopfs.security.kerberos.whatever
hadoopfs.security.kerberos.tokenwhatever

etc.

Reply:
auto-conversion vs custom code comes with pros and cons. I agree with the issues with incompatibility etc but custom code comes with its own set of issues as well. Prefer to stick with well-known libraries for ser/deser.

Yes, yaml supports arbitrary key values. See the sample spec.

Reply:
in yaml you can represent them as  either:
```
hadoopfs.baseurl : xxx
hadoopfs.security.kerberos.whatever: xxx
hadoopfs.security.kerberos.tokenwhatever: xxxx
```
or 
```
hadoopfs:
    baseurl: xxxx
    security:
        kerberos:
            whatever: xxxx
            tokenwhatever: xxxx
```
Both works.
In the map, key names will be assembled using dot.